### PR TITLE
Test signature lost db recreation

### LINKED ISSUE #523: Test signature verification and db recreation on failure
The database signatures are verified on start and the db is recreated if they are corrupted.

Need to add a unit test for this, using a corrupted db file.

### PR DESCRIPTION
**Pull request by gz-c**
Changed the `load()` logic to save a copy of the bad db file as `$file.corrupt.$hash` where hash is the first 8 bytes of the sha1 hash of the file, base64 encoded.  So if we encounter it again we can get a copy.

Also update main README go setup instructions, added table of contents.

Fixes #523

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,30 +17,83 @@ Skycoin is small part of OP Redecentralize and OP Darknet Plan.
...
+### Installing Go
+
+[Install go1.9+](https://golang.org/doc/install)
+
+### Set GOPATH in your environment
+
+Create a folder for your `$GOPATH`, for example:
+
+```sh
+mkdir ~/.gopath
```
Comment (iketheadore) [inline README.md]: why use hiden folder: ~/.gopath, the default GOPATH is $HOME/go since go1.8.

Comment (gz-c) [reply README.md]: oh didn't know they default now

Comment (gz-c) [reply README.md]: you don't need to set gopath anymore?

**Comment by gz-c** [comment]
updated readme

**Pull request merged**